### PR TITLE
Use <SettingTextbox> for the textbox on the dialog of enabling Sync

### PR DESCRIPTION
**Pull request by luixxiul**
For #7152

Based on https://github.com/brave/browser-laptop/pull/6882#issuecomment-276785903

Auditors: @ayumi @bsclifton

Test Plan: n/a

### CODE REVIEW

**Comment by luixxiul** [comment]
The change will make the input form into:

screenshot 2017-02-10 21 14 39

File: less/about/preferences.less [hunk 1]
```diff
@@ -1589,9 +1589,7 @@ table.sortableTable {
   .setupContent {
     margin: 1em 0;
   }
-  .deviceNameInput {
-    width: 50%;
```
Comment (luixxiul) [inline less/about/preferences.less]: Please let me know if this should not be removed.

File: app/renderer/components/preferences/syncTab.js [hunk 2]
```diff
@@ -117,7 +118,9 @@ class SyncTab extends ImmutableComponent {
       : getSetting(settings.SYNC_DEVICE_NAME, this.props.settings)
     return <SettingItem>
       <span data-l10n-id='syncDeviceNameInput' />
-      <input className='deviceNameInput formControl' spellCheck='false'
+      <SettingTextbox
```
Comment (ayumi) [inline app/renderer/components/preferences/syncTab.js]: Earlier I tried to use a Textbox however I couldn't get the prop `ref={}` to work so the input didn't work[0]. I gave up and used a standard input.
This branch doesn't seem to set the device name either.
[0] Maybe this: https://facebook.github.io/react/docs/refs-and-the-dom.html

Test Plan:
- Clear profile
- Enable sync and name the device
- Observe terminal for the device name– should not be "browser-laptop"

Also tested by: `yarn test -- --grep='"^Sync Panel"'`

File: less/forms.less [hunk 3]
```diff
@@ -766,6 +766,7 @@ select {
 }
 
 // From commonStyles.js
+// TODO: remove this after input element in addOverlayContent on syncTab.js is replaced with the new textbox control in Aphrodite
```
Comment (ayumi) [inline less/forms.less]: :100:

File: app/renderer/components/preferences/syncTab.js (hunk 2 above)
Comment (luixxiul) [inline app/renderer/components/preferences/syncTab.js]: @bsclifton @cezaraugusto do you have any idea here?

**Comment by ayumi** [comment]
Hmm, I think as per earlier https://github.com/brave/browser-laptop/pull/7164#discussion_r100586892 this patch breaks naming devices.

The relevant failing test is:
```
npm run test -- --grep='"^Sync Panel sync setup sync profile"'
```

**Comment by ayumi** [comment]
I've reverted this and created https://github.com/brave/browser-laptop/issues/7360 to track the issue.

**Pull request merged**